packages/query/src/block-processor.ts

```diff
@@ -386,12 +386,12 @@ export class BlockProcessor implements BlockProcessorInterface {
   private async assertRootValid(blockHeight: bigint): Promise<void> {
     const sourceOfTruth = await this.querier.cnidarium.keyValue(`sct/anchor/${blockHeight}`);
     const inMemoryRoot = this.viewServer.getSctRoot();
-
-    if (!decodeSctRoot(sourceOfTruth).equals(inMemoryRoot)) {
-      throw new Error(
-        `Block height: ${blockHeight}. Wasm root does not match remote source of truth. Programmer error.`,
-      );
-    }
+    if (!validSctRoot({ inner: sourceOfTruth }))
+      throw new Error(`Block height: ${blockHeight}. Remote cnidarium provided invalid SCT root.`);
+    if (!validSctRoot(inMemoryRoot))
+      throw new Error(`Block height: ${blockHeight}. Invalid local SCT root.`);
+    if (!inMemoryRoot.equals({ inner: sourceOfTruth }))
+      throw new Error(`Block height: ${blockHeight}. Local SCT root does not match remote.`);
```

Comment:
We don't so much want to validate the root, but ensure our locally stored one is the same as the one the remote rpc says. There's also been some breaking changes since we've used this. Going to make some updates here.